packages/babel-plugin-transform-classes/src/transformClass.js

```diff
@@ -371,21 +338,66 @@ export default function transformClass(
   }
 
   function verifyConstructor() {
-    if (!classState.isDerived) return;
+    const { userConstructorPath: path, isDerived, file } = classState;
+    if (!path) return;
 
-    const path = classState.userConstructorPath;
     const body = path.get("body");
 
-    path.traverse(findThisesVisitor);
-
-    let guaranteedSuperBeforeFinish = !!classState.bareSupers.size;
-
+    let thisAlias;
     let thisRef = function() {
-      const ref = path.scope.generateDeclaredUidIdentifier("this");
+      const ref = path.scope.generateUidIdentifier("this");
       thisRef = () => t.cloneNode(ref);
+      path.scope.push({
+        id: ref,
+        init: (thisAlias = !isDerived ? t.thisExpression() : null),
```

Comment:
`assertThisInitialized` return `this` again. What about calling it here?

Example:
```js
var _this = _assertThisInitialized(this);
```

So you don't need to keep track of `assertThisInited`